static-pod-check/README.md

```diff
@@ -0,0 +1,216 @@
+# Static Pod Container Check
+
+- [Static Pod Container Check](#static-pod-container-check)
+  - [Overview](#overview)
+  - [Installation](#installation)
+    - [Ignition Config](#ignition-config)
+    - [Site Config ignitionConfigOverride](#site-config-ignitionconfigoverride)
+    - [MachineConfig](#machineconfig)
+  - [Checking Logs](#checking-logs)
+
+## Overview
+
+The `static-pod-check-workaround.sh` utility monitors the state of static pod containers, triggering a restart of
+`kubelet.service` if any given pod is deemed unhealthy for too long.
+
+```console
+Usage: static-pod-check-workaround.sh
+Options:
+    --dryrun | -n                 Dry run (don't restart kubelet, log only)
+    --verbose | -v                Turn on debug logs
+    --log-to-stdout               Log to stdout instead of journald
+    --manifests-dir | -m          Location of static pod manifests (default: /etc/kubernetes/manifests)
+    --kubelet-up-min <secs>       Minimum time, in seconds, kubelet must be active
+                                  before allowing health checks to run (default: 1200)
+    --healthy-interval <secs>     Interval between checks, in seconds, when static
+                                  pods are healthy (default: 300)
+    --unhealthy-interval <secs>   Interval between checks, in seconds, when static
+                                  pods are unhealthy (default: 30)
+    --max-failures <int>          Maximum consecutive health check failures for any given pod
+                                  before restarting kubelet (default: 20)
+```
+
+Before running any checks on the static pods, the script first verifies that `kubelet.service` has been active for a
+minimum period. This period can be configured using the `--kubelet-up-min` option, with a default of 1200 seconds
+(20 minutes).
+
+Once `kubelet.service` is active for the minimum period, the following checks are performed:
+
+- Query the status of running containers, caching `crictl ps -o json` output for the container checks.
+- Parse the static pod manifests for `system-node-critical` pods to get a list of containers and verify that each
+  container is running.
+- If any given static pod does not have all its containers in a running state, that pod is considered to be unhealthy
+  and a per-pod failure counter is incremented.
+- If any given static pod reaches the maximum number of consecutive health check failures allowed, `kubelet.service`
+  will be restarted. This maximum is configurable with the `--max-failures` option, defaulted to 20 consecutive
+  failures.
+
+If all static pods are healthy, the checks are performed every 300 seconds (configurable with the `--healthy-interval`
+option). If there are unhealthy pods, rechecks are performed every 30 seconds (configurable with the
+`--unhealthy-interval` option).
+
+By default, logs are written to journald with a tag of `static-pod-check-workaround`.
+
+## Installation
```

Comment:
How can this workaround be removed or disabled? Should that be mentioned in the README as well?

Reply:
I'll add a section

Reply:
Looks good